src/constraint/SequentialSampleable.cpp

```diff
@@ -0,0 +1,132 @@
+#include "aikido/constraint/SequentialSampleable.hpp"
+
+#include <dart/common/StlHelpers.hpp>
+
+namespace aikido {
+namespace constraint {
+
+//==============================================================================
+class SequentialSampleGenerator : public SampleGenerator
+{
+public:
+  SequentialSampleGenerator(
+      statespace::StateSpacePtr stateSpace,
+      std::vector<std::unique_ptr<SampleGenerator>>& generators)
+    : mStateSpace(std::move(stateSpace))
+    , mGenerators(generators)
+    , mIndex(0)
+  {
+    for (std::size_t i = 0; i < mGenerators.size(); ++i)
+      assert(mGenerators[i]->getStateSpace() == mStateSpace);
+  }
+
+  // Documentation inherited
+  statespace::StateSpacePtr getStateSpace() const override
+  {
+    return mStateSpace;
+  }
+
+  // Documentation inherited
+  bool sample(statespace::StateSpace::State* state)
+  {
+    while (canSample())
+    {
+      if (mGenerators[mIndex]->canSample())
+        return mGenerators[mIndex]->sample(state);
+
+      mIndex++;
+    }
+    return false;
+  }
+
+  // Documentation inherited
+  int getNumSamples() const override
+  {
+    if (mGenerators.empty())
```

Comment:
I think this might be unnecessary. If this happens, the for loop wouldn't execute and `numSamples` would be 0 anyway.